packages/fuel-gauge/src/transaction.test.ts

```diff
@@ -140,3 +154,167 @@ describe('Transaction', () => {
+        args: ['--poa-instant', 'false', '--poa-interval-period', '1ms'],
+        loggingEnabled: false,
+        snapshotConfig,
+      },
+    });
+    privileged.provider = provider;
+
+    return { provider, privileged, wallets, cleanup, [Symbol.dispose]: cleanup };
+  };
+
+  const upgradeConsensusParameters = async (wallet: WalletUnlocked, bytecode: BytesLike) => {
+    const request = new UpgradeTransactionRequest();
+    request.addConsensusParametersUpgradePurpose(bytecode);
+
+    const cost = await wallet.getTransactionCost(request);
+    request.maxFee = cost.maxFee;
+    await wallet.fund(request, cost);
+
+    const response = await wallet.sendTransaction(request);
+    return response.waitForResult();
+  };
+
+  it('should correctly update the privileged address in consensus data', async () => {
+    // Consensus parameters with other privileged address
+    const CONSENSUS_BYTECODE = decompressBytecode(
+      'H4sIAAAAAAAAA2Ng+M8Igg2Hrus3NAg0MDE0LGCDIqAQlMcAlHrAyxBjz8DMQAvASFUEwj9+bNOwrV+6rWirwekd/mdO+/+44/BDPO9LiO9tecmgtewgz035fyZ4x6QMr3O/Xp9YfGXp1KcaK86n+b22+5wvoLChYS4vAOSrKjscAQAA'
+    );
+
+    using launched = await setupTestNode();
+    const { privileged } = launched;
+
+    // Upgrade the privileged address
+    const { isStatusSuccess, isTypeUpgrade } = await upgradeConsensusParameters(
+      privileged,
+      CONSENSUS_BYTECODE
+    );
+    expect(isStatusSuccess).toBeTruthy();
+    expect(isTypeUpgrade).toBeTruthy();
+
+    // Attempts to update again but the privileged address has been changed
+    await expect(() =>
+      upgradeConsensusParameters(privileged, CONSENSUS_BYTECODE)
+    ).rejects.toThrowError(/Validity\(TransactionUpgradeNoPrivilegedAddress\)/);
+  });
+
+  it('should correctly update the gas costs in consensus data', async () => {
+    // Consensus parameters bytecode with gas costs free
+    const CONSENSUS_BYTECODE = decompressBytecode(
+      'H4sIAAAAAAAAA2Ng+M8Igg3ty/kaGgQamBgaFrBBEVAIymMASj3gZXhwjtWegZmBFoCRqgiE57zYxuf8JLNhe+05Zy2x8zc2xvTvl8iSn3fzmHe0nKOvG8h7Qbt2f1V/ZP/1yU4lfhHFKe81Hpe6mz+Q1a/42P4g02KvOQDW/Lz1HgEAAA=='
+    );
+
+    using launched = await setupTestNode();
+    const { privileged, provider } = launched;
+
+    const {
+      chain: { consensusParameters: consensusBeforeUpgrade },
+    } = await provider.fetchChainAndNodeInfo();
+
+    // Update the gas costs to free
+    const { isStatusSuccess, isTypeUpgrade } = await upgradeConsensusParameters(
+      privileged,
+      CONSENSUS_BYTECODE
+    );
+    expect(isStatusSuccess).toBeTruthy();
+    expect(isTypeUpgrade).toBeTruthy();
+
+    // Fetch the upgraded gas costs, they should be different from before
+    const {
+      chain: { consensusParameters: consensusAfterUpgrade },
+    } = await provider.fetchChainAndNodeInfo();
+
+    expect(consensusBeforeUpgrade.gasCosts).not.toEqual(consensusAfterUpgrade.gasCosts);
+  });
+
+  it(
+    'should load bytecode in subsections slowly',
+    async () => {
```

Comment:
Moving timeout up here. ([example](https://github.com/FuelLabs/fuels-ts/blob/b8b5fa04fc1f1bdbda1a8950cedff589b2b29975/packages/fuel-gauge/src/contract-factory.test.ts#L295-L298))

Would 10s or 15s be enough?

```suggestion
    { timeout: 10000 },
    async () => {
```

Reply:
Probably not, I changed it to 30s.

Reply:
Looks like 30s is turning out to be too less as well.

Reply:
Hey guys, I isolated the upgrade/upload transaction tests into a separate file to better reuse the information before and after the tests run. I also divided the subsections into larger parts to reduce execution time.